### PR TITLE
Amp-16 editors pick component

### PR DESCRIPTION
**Pull request by wajerrr**
- updated Graphic QL query 
 - added story collection component

I have divided them into smaller logical components

I have put them in `story-collection` folder as there is 10 files (including unit test files) @umbobabo @dosyara @yasgreen93  is this approach ok ? or would you prefer to have flat component folder structure.

### CODE REVIEW

File: src/app/components/story-collection/story-collection-item-headline.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+`;
+
+const ItemHeadline = ({
+  subHeadline,
+  subHeadlineClassName,
+  headlineClassName,
+  children,
+  className,
+}) => (
+  <h3 className={`${header} ${className}`}>
```
Comment (yasgreen93) [inline src/app/components/story-collection/story-collection-item-headline.js]: This way of doing styles is cool, I guess it's now down to everyone's preferences??

File: src/app/components/story-collection/story-collection-hero-item.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import text from '../../styles/typography';
+import spacings from '../../styles/spacings';
+import SectionLink from './story-collection-section-link';
+import ItemHeadline from './story-collection-item-headline';
+import color from '../../styles/color';
+import fontFamily from '../../styles/font-family';
+
+const description = css`
+  margin: 0;
+  padding: 0;
```
[the remark this answers is not in the document]
Comment (yasgreen93) [reply src/app/components/story-collection/story-collection-hero-item.js]: Just looked through all of it and there is this file https://github.com/EconomistDigitalSolutions/amp/blob/master/src/app/styles/spacings.js which has `none: 0`.
If we're just writing `0` everywhere do we need that var?

File: src/app/components/story-collection/story-collection-item.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import text from '../../styles/typography';
+import spacings from '../../styles/spacings';
+import SectionLink from './story-collection-section-link';
+import ItemHeadline from './story-collection-item-headline';
+
+const textContainer = css`
+  padding: ${spacings.s};
+  width: 50%;
+`;
+
```
Comment (yasgreen93) [inline src/app/components/story-collection/story-collection-item.js]: Again this is personal preference but I remember @umbobabo commenting on my PR about spaces between style consts. My PR doesn't have any spaces, do we want to decided on spaces or no spaces? Very small and unnecessary thing I know...

File: src/app/components/story-collection/story-collection-item.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+const link = css`
+  display: flex;
+`;
+
+const StoryCollectionItem = ({ data, className }) => (
+  <li className={className}>
+    <a className={link} href={data.url.canonical}>
```
Comment (yasgreen93) [inline src/app/components/story-collection/story-collection-item.js]: Once my PR is merged you could use `<StyledLink />`

Comment (yasgreen93) [reply src/app/components/story-collection/story-collection-item.js]: Or thinking about what @dosyara said either use the StyledLink component or create a new one specifically for this story collection to have more context?

File: src/app/components/story-collection/story-collection.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { css } from 'emotion';
+import { PropTypes } from 'prop-types';
+import StoryCollectionItem from './story-collection-item';
+import StoryCollectionHeroItem from './story-collection-hero-item';
+import spacings from '../../styles/spacings';
```
Comment (yasgreen93) [inline src/app/components/story-collection/story-collection.js]: Some files you have spaces in between imports. Shall we make this consistent?

File: src/server/get-graphql-data.js [hunk 6]
```diff
@@ -34,9 +34,41 @@ const client = new ApolloClient({
 const getGraphqlData = async (ref) => {
   const query = gql`
         {
-          canonical(ref: "${ref}") {
+          canonical :canonical(ref: "${ref}") {
             ...C
+          },
+          editorsPick: canonical(ref: "/content/9qe6f6cm77btf0phaepjui01ckh6rfpu") {
```
Comment (yasgreen93) [inline src/server/get-graphql-data.js]: We're using a fragment for the canonical article query but then this style for editors picks. I think we could re-use some of the query. In fe-blogs Ellis wrote a `baseUniversalQuery` and other fragments which you can slot in when needed. Could we do something like that?
https://github.com/EconomistDigitalSolutions/fe-blogs/blob/master/src/app/composition/index.js#L41

File: src/app/components/story-collection/story-collection-hero-item.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (umbobabo) [reply src/app/components/story-collection/story-collection-hero-item.js]: That's spacing for typography only

[the remark this answers is not in the document]
Comment (wajerrr) [reply src/app/components/story-collection/story-collection-hero-item.js]: well I have changed zeros to`${spacings.none}`

File: src/app/components/story-collection/story-collection-item-headline.js (hunk 1 above)
Comment (wajerrr) [reply src/app/components/story-collection/story-collection-item-headline.js]: well @dosyara will present proposal hot to unify all our approaches

File: src/app/components/story-collection/story-collection-item.js (hunk 3 above)
Comment (wajerrr) [reply src/app/components/story-collection/story-collection-item.js]: well I think it makes it more readable... i don't mind either way

File: src/app/components/story-collection/story-collection.js (hunk 5 above)
Comment (wajerrr) [reply src/app/components/story-collection/story-collection.js]: I generally add line break between external imports from node_modules and project file imports

File: src/server/get-graphql-data.js (hunk 6 above)
Comment (wajerrr) [reply src/server/get-graphql-data.js]: but this fragment gives us more data when we need for editors picks. I have added only properties we need for editors picks

File: src/app/components/story-collection/story-collection-hero-item.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+`;
+
+const headline = css`
+  font-size: ${text.sizeStep['1']};
+  font-weight: 500;
+  letter-spacing: ${text.lineHeight.letterSpacing.serifOnStep['1']};
+  line-height: ${text.lineHeight.serifOnStep['1']};
+`;
+
+const StoryCollectionHeroItem = ({ data, className }) => (
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/app/components/story-collection/story-collection-hero-item.js]: yep, makes sense, done

File: src/app/components/story-collection/story-collection-hero-item.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  <li className={className}>
+    <a href={data.url.canonical}>
+      <amp-img
+        alt={data.channel.headline}
+        layout="responsive"
+        width={data.image.main.width || '1280'}
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/app/components/story-collection/story-collection-hero-item.js]: from my observation that all the images so far have the same dimensions (as far as I know `amp-img` uses it only to find out aspect ratio) but that was just assumption.. removed it for now

File: src/app/components/story-collection/story-collection-item-headline.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+import color from '../../styles/color';
+import fontFamily from '../../styles/font-family';
+import spacings from '../../styles/spacings';
+
+const baseSubHeadline = css`
+  overflow: hidden;
+  display: block;
+  font-weight: normal;
+  margin: 0 0 ${spacings.xs} 0;
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/app/components/story-collection/story-collection-item-headline.js]: done

File: src/app/components/story-collection/story-collection-item-headline.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+import { PropTypes } from 'prop-types';
+
+import color from '../../styles/color';
+import fontFamily from '../../styles/font-family';
+import spacings from '../../styles/spacings';
+
+const baseSubHeadline = css`
+  overflow: hidden;
+  display: block;
+  font-weight: normal;
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/app/components/story-collection/story-collection-item-headline.js]: possibly not, removed, my bad

File: src/app/components/story-collection/story-collection-item-headline.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  margin: 0 0 ${spacings.xs} 0;
+  color: ${color.economist};
+  font-family: ${fontFamily.sans};
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/app/components/story-collection/story-collection-item-headline.js]: not any more font-wight has been removed :-)

File: src/app/components/story-collection/story-collection-item-headline.js [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+`;
+
+const baseHeadline = css`
+  display: block;
+  margin: 0 0 ${spacings.m} 0;
```
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/app/components/story-collection/story-collection-item-headline.js]: done

**Comment by wajerrr** [comment]
@umbobabo @yasgreen93 @dosyara I think I have addressed all the comments

**Pull request merged**